src/nbl/video/CVulkanConnection.h

```diff
@@ -3,43 +3,83 @@
 
 #include <volk.h>
 #include "nbl/video/IAPIConnection.h"
-#include "nbl/video/surface/CSurfaceVKWin32.h"
 #include "nbl/video/CVulkanPhysicalDevice.h"
 
-namespace nbl {
-namespace video
+#if defined(_NBL_PLATFORM_WINDOWS_)
+#   include "nbl/ui/IWindowWin32.h"
+#   include "nbl/video/surface/CSurfaceVKWin32.h"
+#endif
+
+namespace nbl::video
 {
 
 class CVulkanConnection final : public IAPIConnection
 {
 public:
-    CVulkanConnection(uint32_t appVer, const char* appName)
+    CVulkanConnection(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t appVer,
+        const char* appName, const SDebugCallback& dbgCb) : IAPIConnection(std::move(sys))
     {
         VkResult result = volkInitialize();
         assert(result == VK_SUCCESS);
+        
+        VkApplicationInfo applicationInfo = { VK_STRUCTURE_TYPE_APPLICATION_INFO };
+        applicationInfo.apiVersion = VK_MAKE_VERSION(1, 1, 0);
+        applicationInfo.engineVersion = NABLA_VERSION_INTEGER;
+        applicationInfo.pEngineName = "Nabla";
+        applicationInfo.applicationVersion = appVer;
+        applicationInfo.pApplicationName = appName;
+        
+        // Todo(achal): Get this from the user
+        const std::vector<const char*> requiredInstanceLayerNames({ "VK_LAYER_KHRONOS_validation" });
+        assert(areAllInstanceLayersAvailable(requiredInstanceLayerNames));
+
+        // Note(achal): This exact create info is used for application wide debug messenger for now
+        VkDebugUtilsMessengerCreateInfoEXT debugUtilsMessengerCreateInfo = { VK_STRUCTURE_TYPE_DEBUG_UTILS_MESSENGER_CREATE_INFO_EXT };
+        debugUtilsMessengerCreateInfo.messageSeverity = VK_DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT;
+        debugUtilsMessengerCreateInfo.messageType = VK_DEBUG_UTILS_MESSAGE_TYPE_GENERAL_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_TYPE_VALIDATION_BIT_EXT
+            | VK_DEBUG_UTILS_MESSAGE_TYPE_PERFORMANCE_BIT_EXT;
```

Comment:
@sadiuk think about what methods and fields to give `ILogger` so these bitflags can be deduced